graphix/visualization.py

```diff
@@ -796,9 +791,8 @@ def get_pos_wo_structure(self) -> dict[int, _Point]:
             pos[node][0] = l_max - layer
         vert = list({pos[node][1] for node in self.graph.nodes()})
         vert.sort()
-        for node in self.graph.nodes():
-            pos[node][1] = vert.index(pos[node][1])
-        return pos
+        index = {y: i for i, y in enumerate(vert)}
+        return {node: (pos[node][0], index[pos[node][1]]) for node in self.graph.nodes()}
```

Comment:
Here we can do as you did in L705 and remove the loop in L790, right?

```suggestion
        return {node: (l_max - layer, index[pos[node][1]]) for node, layer in layers.items()}
```